### PR TITLE
Add option to normalize HTTPS to HTTP

### LINKED ISSUE #34: Normalize HTTPS to HTTP
Currently http://wesbite.com
and https://website.com are treated differently.

### PR DESCRIPTION
**Pull request by ScionOfBytes**
Fixes #34.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -52,6 +52,21 @@ normalizeUrl('//sindresorhus.com:80/', {normalizeProtocol: false});
 //=> '//sindresorhus.com'
 ```
 
+##### normalizeSsl
```
Comment (sindresorhus) [inline readme.md]: I think `normalizeHttps` would be a better name.

File: readme.md [hunk 2]
```diff
@@ -52,6 +52,21 @@ normalizeUrl('//sindresorhus.com:80/', {normalizeProtocol: false});
...
+
+Type: `boolean`<br>
+Default: `false`
+
+Normalize `https:` URLs to `http`.
```
Comment (sindresorhus) [inline readme.md]: `http` => `http:`

File: index.js [hunk 3]
```diff
@@ -153,5 +154,11 @@ module.exports = function (str, opts) {
 		str = str.replace(/^http:\/\//, '//');
 	}
 
+	var hasHttps = str.indexOf('https') === 0;
+
+	if (opts.normalizeSsl && hasHttps) {
+		str = str.replace(/^https/, 'http');
+	}
```
Comment (sindresorhus) [inline index.js]: Would be better to work on the parsed URL object further up in the code, so you could just replace `urlObj.protocol`.

File: index.js [hunk 4]
```diff
@@ -53,6 +54,12 @@ module.exports = function (str, opts) {
 
 	var urlObj = url.parse(str);
 
+	var hasHttps = urlObj.protocol === 'https:';
```
Comment (ScionOfBytes) [inline index.js]: @sindresorhus Please check.

File: readme.md [hunk 5]
```diff
@@ -52,6 +52,21 @@ normalizeUrl('//sindresorhus.com:80/', {normalizeProtocol: false});
...
+Type: `boolean`<br>
+Default: `false`
+
+Normalize `https:` URLs to `http:`.
+
+```js
+normalizeUrl('https://sindresorhus.com:80/');
+//=> 'https://sindresorhus.com'
+
+normalizeUrl('https://sindresorhus.com:80/', {normalizeSsl: true});
```
Comment (sindresorhus) [inline readme.md]: Option name is outdated

File: readme.md (hunk 1 above)
Comment (sindresorhus) [inline readme.md]: Needs to be updated too

File: index.js (hunk 4 above)
Comment (sindresorhus) [inline index.js]: Just put this inline in the if-statement.

**Comment by ScionOfBytes** [comment]
@sindresorhus Made the updates.

File: test.js [hunk 6]
```diff
@@ -57,6 +57,14 @@ test('removeQueryParameters option', t => {
 	t.is(m('www.sindresorhus.com?foo=bar&utm_medium=test&ref=test_ref', opts), 'http://www.sindresorhus.com/?foo=bar');
 });
 
+test('normalizeHttps option', t => {
+	const opts = {normalizeHttps: true};
+
+	t.is(m('http://sindresorhus.com'), 'http://sindresorhus.com');
```
Comment (shinnn) [inline test.js]: Missing `, opts`?

Comment (ScionOfBytes) [inline test.js]: @shinnn No, more to show that if the option is not given, nothing changes. I'll updated this to make more sense.

**Comment by ScionOfBytes** [comment]
@shinnn See if these make more sense.

File: test.js (hunk 6 above)
Comment (shinnn) [inline test.js]: You're right. Other option tests also have the same lines for clarification. I should have read entire code, not just PR diff.

**Pull request merged**